addons/misc/functions/fnc_suitcaseNukeZeus.sqf

```diff
@@ -0,0 +1,211 @@
+#include "script_component.hpp"
+/*/////////////////////////////////////////////////
+Author: Landric
+               
+File: fnc_suitcaseNukeZeus.sqf
+Parameters: pos
+Return: none
+
+*///////////////////////////////////////////////
+params [["_pos",[0,0,0],[[]],3]];
+
+private _onConfirm =
+{
+    params ["_dialogResult","_in"];
+    _in params [["_pos",[0,0,0],[[]],3]];
+    _dialogResult params [
+        "_countdownLength",
+        "_effect",
+        "_defuser",
+        "_defuseTime",
+        ["_code", "", [""]],
+        ["_target", 0, [0]]
+    ];
+
+    private _device = "Land_MultiScreenComputer_01_black_F" createVehicle _pos;
+    ["zen_common_updateEditableObjects", [[_device]]] call CBA_fnc_serverEvent;
+    _device setVariable [QGVAR(suitcaseNuke_isArmed), true, true];
+
+
+    // Timer code
+    // TODO: move to server to avoid problems if zeus crashes?
+    [_device, _countdownLength] spawn { 
```

Comment:
Would it be possible to avoid using scheduled environment? I am trying to avoid it unless its for visual particle effects etc. 
As you know it have issues with your loop and code can be cut off mid execution due to scheduler. 

I was thinking if something like an PFEH or waitUntilAndExecute from CBA would be better? As it seems you don't really use sleep outside slowing the loops down a bit. 

Instead you could do a PFEH with execution every 100ms (or a fitting value) that would then

1. Update remaining time
2. Check if time hit 0, then stop PFEH and handle effect
3. Check if disarmed, then halt stop the PFEH

It will probably make it a bit cleaner code-wise too, as you just have a single loop, instead of multiple while-loops right after each other. 

https://cbateam.github.io/CBA_A3/docs/files/common/fnc_addPerFrameHandler-sqf.html
https://cbateam.github.io/CBA_A3/docs/files/common/fnc_waitUntilAndExecute-sqf.html  

https://github.com/CBATeam/CBA_A3/wiki/Per-Frame-Handlers

Reply:
I'm not sure it'd be possible to (cleanly) do it in a single loop - the different loops are because the texture needs to refresh at different frequencies before and after the device is triggered. But I can give it a look!

Reply:
You can make a loop running at fast enough for the fastest updating. Then use if loops checking time-delta since last loop to see if it should trigger.
One of the usual patterns on embedded devices. 

Psudo code, assuming time is given in seconds
```cpp
float last_time = 0;
loop {
  float time_diff = TimeNow() - last_time;
  
  // 5hz
  if (time_diff > 0.2 and device_triggered) then {
    show_screen_effect_triggered();
  }

  // 2hz
  if (time_diff > 0.5 and !device_triggered ) then  {
    show_countdown_effect();
  }

  last_time = now(); 
}
```

Might want to use lazy eval and check the bool first, but should be negligible.

Reply:
Oh, I guess so! I was trying to avoid "needlessly" checking conditions, but the overhead is hardly going to be huge, and if it makes the code cleaner, much better to do it that way 👍